tests/src/test/scala/com/nvidia/spark/rapids/spill/SpillFrameworkSuite.scala

```diff
@@ -0,0 +1,1103 @@
+        assertResult(false)(handle.spillable)
+        assertResult(0)(SpillFramework.stores.hostStore.spill(sizeOnHost))
+        // leak it by increasing the ref count of the underlying cuDF column
+        RapidsHostColumnVector.extractBases(cb).head.incRefCount()
+      }
+      withResource(leakedFirstColumn) { _ =>
+        // 0 because we have a reference to the first column
+        assertResult(false)(handle.spillable)
+        assertResult(0)(SpillFramework.stores.hostStore.spill(sizeOnHost))
+      }
+      // batch is now spillable because we close our reference to the column
+      assertResult(true)(handle.spillable)
+      assertResult(sizeOnHost)(SpillFramework.stores.hostStore.spill(sizeOnHost))
+    }
+  }
+
+  test("host originated: an aliased host batch is not spillable (until closing the original) ") {
+    val (hostBatch, sparkTypes) = buildHostBatch()
+    val handle = SpillableHostColumnarBatchHandle(hostBatch)
+    withResource(handle) { _ =>
+      assertResult(1)(SpillFramework.stores.hostStore.numHandles)
+      assertResult(true)(handle.spillable)
+      withResource(handle.materialize(sparkTypes)) { _ =>
+        assertResult(false)(handle.spillable)
+      } // we now have two copies in the store
+      assertResult(true)(handle.spillable)
+    }
+  }
+
+  test("host originated: a host batch supports duplicated columns") {
+    val (hostBatch, sparkTypes) = buildHostBatchWithDuplicate()
+    val handle = SpillableHostColumnarBatchHandle(hostBatch)
+    withResource(handle) { _ =>
+      assertResult(1)(SpillFramework.stores.hostStore.numHandles)
+      assertResult(true)(handle.spillable)
+      withResource(handle.materialize(sparkTypes)) { _ =>
+        assertResult(false)(handle.spillable)
+      } // we now have two copies in the store
+      assertResult(true)(handle.spillable)
+    }
+  }
+
+  test("host originated: a host batch supports aliasing and duplicated columns") {
+    SpillFramework.shutdown()
+    val sc = new SparkConf
+    // disables the host store limit
+    sc.set(RapidsConf.OFF_HEAP_LIMIT_ENABLED.key, "true")
+    SpillFramework.initialize(new RapidsConf(sc))
+
+    try {
+      val (hostBatch, sparkTypes) = buildHostBatchWithDuplicate()
+      withResource(SpillableHostColumnarBatchHandle(hostBatch)) { handle =>
+        withResource(SpillableHostColumnarBatchHandle(handle.materialize(sparkTypes))) { handle2 =>
+          assertResult(2)(SpillFramework.stores.hostStore.numHandles)
+          assertResult(false)(handle.spillable)
+          assertResult(false)(handle2.spillable)
+        }
+        assertResult(true)(handle.spillable)
+      }
+    } finally {
+      SpillFramework.shutdown()
+    }
+  }
+
+  // this is a key behavior that we wanted to keep during the spill refactor
+  // where host objects that are added directly to the store do not cause a
+  // host->disk spill on their own, instead they will get spilled later
+  // due to device->host spills.
```

Comment:
Is this because we are assuming the memory is already allocated/accounted for? Are we still adding the new buffer size to the total store allocated size?

Reply:
yes and yes. The host memory `was` allocated,  and we track it as such. But we are not going to actively spill right now. We'll spill later when device wants to spill.

This may change later. And when host limits are enabled (not host store limits) then it fits that model better.